### ISSUE TITLE
Card Open Webhook Error

### ISSUE THREAD

**Issue by jymcheong** [wekan/wekan#2780]
## Version
v3.49

## Issue

The older version of Wekan had no problem with Card click outgoing webhook. Recently I git clone the latest version for a test. `docker logs wekan-app` shows the following:  

```

Exception while invoking method 'outgoingWebhooks' { [object Object]
  message: 'Match error: Expected plain object',
  path: '',
  sanitizedError: 
   { Error: Match failed [400]
    at errorClass.<anonymous> (packages/check/match.js:91:27)
    at new errorClass (packages/meteor.js:655:17)
    at check (packages/check/match.js:36:17)
    at MethodInvocation.outgoingWebhooks (server/notifications/outgoing.js:103:5)
    at MethodInvocation.methodMap.(anonymous function) (packages/mdg:meteor-apm-agent/lib/hijack/wrap_session.js:161:30)
    at currentArgumentChecker.withValue (packages/check/match.js:118:15)
    at Meteor.EnvironmentVariable.EVp.withValue (packages/meteor.js:1234:12)
    at Object._failIfArgumentsAreNotAllChecked (packages/check/match.js:116:43)
    at maybeAuditArgumentChecks (packages/ddp-server/livedata_server.js:1768:18)
    at DDP._CurrentMethodInvocation.withValue (packages/ddp-server/livedata_server.js:719:19)
    at Meteor.EnvironmentVariable.EVp.withValue (packages/meteor.js:1234:12)
    at DDPServer._CurrentWriteFence.withValue (packages/ddp-server/livedata_server.js:717:46)
    at Meteor.EnvironmentVariable.EVp.withValue (packages/meteor.js:1234:12)
    at Promise (packages/ddp-server/livedata_server.js:715:46)
    at new Promise (<anonymous>)
    at Session.method (packages/ddp-server/livedata_server.js:689:23)
    at packages/mdg:meteor-apm-agent/lib/hijack/wrap_session.js:52:38
    at Meteor.EnvironmentVariable.EVp.withValue (packages/meteor.js:1234:12)
    at Session.sessionProto.protocol_handlers.method (packages/mdg:meteor-apm-agent/lib/hijack/wrap_session.js:51:44)
    at packages/ddp-server/livedata_server.js:559:43
     isClientSafe: true,
     error: 400,
     reason: 'Match failed',
     details: undefined,
     message: 'Match failed [400]',
     errorType: 'Meteor.Error' },
  errorType: 'Match.Error' }

```

**Comment by xet7** [comment]
Please test with docker version tags, what was the version where it worked, and then in next version it did not work anymore.

**Comment by xet7** [comment]
Similar object error at https://github.com/wekan/wekan/issues/2619#issuecomment-548607477

**Comment by jymcheong** [comment]
> Please test with docker version tags, what was the version where it worked, and then in next version it did not work anymore.

Noted. Will do that in the future.

I narrowed it down to:

![Screenshot 2019-11-04 17 51 33](https://user-images.githubusercontent.com/11232021/68112775-1b692e80-ff2d-11e9-9131-442039333b1c.png)

Will modify CardDetails.js at my fork to test tomorrow. Knocking off soon.

**Comment by xet7** [comment]
That is Global Webhooks feature https://github.com/wekan/wekan/pull/2665 by @whowillcare .

Please look how to fix this so that both Global Webhooks and Card Open Webhook works.

**Comment by whowillcare** [comment]
I'll take a look when I have time

**Comment by xet7** [comment]
Related #2666

**Comment by xet7** [comment]
Open webhook was originally added at [Wekan v3.18](https://github.com/wekan/wekan/blob/master/CHANGELOG.md#v318-2019-08-15-wekan-release) with this https://github.com/wekan/wekan/pull/2620

**Comment by jymcheong** [comment]
> Related #2666

For #2666, events are sent to webhook receiver service even when CARD_OPENED_WEBHOOK_ENABLED=false

For this case, it is the opposite such that no webhook event was received due to the changes with Global webhook. I modified cardDetails.js such that it is working again but that would mean #2666 will get the CardSelected events again if merged. 

For some strange reason, when I ran Wekan with Meteor directly, it honours CARD_OPENED_WEBHOOK_ENABLED=false & no hook events for card selected.

But when I I built a Docker Image (with modified cardDetails.js) with DockerFile which has a CARD_OPENED_WEBHOOK_ENABLED=false line in it & modified docker-compose.yml to use my custom image instead of quay.io's , it will still emit CardSelected even when option is set to false.
